pkg/utils/data_format.go

```diff
@@ -29,6 +30,74 @@ var (
 	Q = math.Pow(10, 15)
 )
 
+func SecondsToHuman(input int) (result string) {
+	years := math.Floor(float64(input) / 60 / 60 / 24 / 7 / 30 / 12)
+	seconds := input % (60 * 60 * 24 * 7 * 30 * 12)
+	months := math.Floor(float64(seconds) / 60 / 60 / 24 / 7 / 30)
+	seconds = input % (60 * 60 * 24 * 7 * 30)
+	weeks := math.Floor(float64(seconds) / 60 / 60 / 24 / 7)
+	seconds = input % (60 * 60 * 24 * 7)
+	days := math.Floor(float64(seconds) / 60 / 60 / 24)
+	seconds = input % (60 * 60 * 24)
+	hours := math.Floor(float64(seconds) / 60 / 60)
+	seconds = input % (60 * 60)
+	minutes := math.Floor(float64(seconds) / 60)
+	seconds = input % 60
+
+	if years > 0 {
+		result = fmt.Sprintf("%d y %d m %d w %d d %d H %d M %d S",
+			int(years),
+			int(months),
+			int(weeks),
+			int(days),
+			int(hours),
+			int(minutes),
+			int(seconds),
+		)
+	} else if months > 0 {
+		result = fmt.Sprintf("%d m %d w %d d %d H %d M %d S",
+			int(months),
+			int(weeks),
+			int(days),
+			int(hours),
+			int(minutes),
+			int(seconds),
+		)
+	} else if weeks > 0 {
+		result = fmt.Sprintf("%d w %d d %d H %d M %d S",
+			int(weeks),
+			int(days),
+			int(hours),
+			int(minutes),
+			int(seconds),
+		)
+	} else if days > 0 {
+		result = fmt.Sprintf("%d d %d H %d M %d S",
+			int(days),
+			int(hours),
+			int(minutes),
+			int(seconds),
+		)
+	} else if hours > 0 {
+		result = fmt.Sprintf("%d H %d M %d S",
+			int(hours),
+			int(minutes),
+			int(seconds),
+		)
+	} else if minutes > 0 {
+		result = fmt.Sprintf("%d M %d S",
+			int(minutes),
+			int(seconds),
+		)
+	} else {
+		result = fmt.Sprintf("%d S",
+			int(seconds),
+		)
+	}
```

Comment:
On a second thought, `4d 1h 0m 24s` looks even better (all small letters, no spaces). I noticed that you used caps to differentiate between month and minute, like in most date time format specifiers. But why not use `M` for month and `m` for minute?

Ik this is a small thing, but what do you think?

```suggestion
	if years > 0 {
		result = fmt.Sprintf("%dY %dM %dw %dd %dh %dm %ds",
			int(years),
			int(months),
			int(weeks),
			int(days),
			int(hours),
			int(minutes),
			int(seconds),
		)
	} else if months > 0 {
		result = fmt.Sprintf("%dM %dw %dd %dh %dm %ds",
			int(months),
			int(weeks),
			int(days),
			int(hours),
			int(minutes),
			int(seconds),
		)
	} else if weeks > 0 {
		result = fmt.Sprintf("%dw %dd %dh %dm %ds",
			int(weeks),
			int(days),
			int(hours),
			int(minutes),
			int(seconds),
		)
	} else if days > 0 {
		result = fmt.Sprintf("%dd %dh %dm %ds",
			int(days),
			int(hours),
			int(minutes),
			int(seconds),
		)
	} else if hours > 0 {
		result = fmt.Sprintf("%dh %dm %ds",
			int(hours),
			int(minutes),
			int(seconds),
		)
	} else if minutes > 0 {
		result = fmt.Sprintf("%dm %ds",
			int(minutes),
			int(seconds),
		)
	} else {
		result = fmt.Sprintf("%ds",
			int(seconds),
		)
	}
```

Reply:
UNIX time format standards are `m` for month and `M` for minute, I didn't want to change that. I've followed the format from the `date` command. [Man page](https://man7.org/linux/man-pages/man1/date.1.html) for the same.

As for the spacing, that does make sense, it looks better. I'll change that.

Reply:
Ah okay, TIL